syntax = "proto2";

package innervoting;

import "google/protobuf/timestamp.proto";

service eVotingReplica {
    rpc ElectionReplica (Election) returns (Status);
    rpc CastVoteReplica (Vote) returns (Status);
}

message Voter {
    required string name = 1;
    required string group = 2;
}
message Status {
    required int32 code = 1;
}

message Election {
    required string name = 1;
    repeated string groups = 2;
    repeated string choices = 3;
    required google.protobuf.Timestamp end_date = 4;
}
message Vote {
    required string election_name = 1;
    required string choice_name = 2;
}

message VoteCount {
    required string choice_name = 1;
    required int32 count = 2;
}

message ElectionStatus {
    required string name = 1;
    repeated string groups = 2;
    repeated string choices = 3;
    repeated VoteCount count = 4;
    repeated string voters = 5;
    required google.protobuf.Timestamp end_date = 6;
}

